Fix of issue NuGet/Home#1802: Pre select packages in consolidated tab.

feiling opened the pull request:
In fact, with this fix, we do auto select in both the consolidated and the updates tabs.

@yishaigalatzer @emgarten @deepakaravindr

Commit:
Fix of issue NuGet/Home#1802: Pre select packages in consolidated tab.

In fact, with this fix, we do auto select in both the consolidated and the updates tabs.

feiling commented on src/NuGet.Clients/PackageManagement.UI/PackageSolutionDetailControlModel.cs:
```diff
@@ -412,6 +413,34 @@ private static bool IsInstalled(NuGetProject project, string id)
+            }
+        }
+
+        public override void OnFilterChanged(Filter filter)
```
we need to clear selections when the current tab is switched to "Browser" or "install".

emgarten commented:
:shipit:

yishaigalatzer commented:
@feiling I'd like to see a short demo

Commit:
The project selections are cleared when user switches from UpdatesAvailable/Consolidate to Browse/Installed.

yishaigalatzer commented on src/NuGet.Clients/PackageManagement.UI/Xamls/PackageManagerTopPanel.xaml.cs:
```diff
@@ -200,4 +201,17 @@ public void SelectFilter(Filter selectedFilter)
+            get;
+        }
+
+        public FilterChangedEventArgs(Filter? previousFilter)
```
This confused me a bit, we could have used a Filter.None instead, but I guess this works and might actually be cleaner.

yishaigalatzer commented:
:shipit:

yishaigalatzer commented:
@feiling does selection change after we install a package say from powershell console?

Commit:
Move AutoSelectProjects() into UpdateInstalledVersions().
This makes sure that selections are updated after projects are added, and after packages
are installed/uninstalled.

feiling commented:
Unfortunately no. Actions in powershell won't cause UI to refresh. This is a known issue.

feiling commented:
Fix merged.
